File: Models/Registry.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace think_agro_metrics.Models
{
    public abstract class Registry
    {
        public long RegistryID { get; set; }
        public long IndicatorID { get; set; }
        public string Name { get; set; }
        public DateTime DateAdded { get; set; } // Date in which the registry is added
        public DateTime Date { get; set; } // Date when the registry occured
        public ICollection<Document> Documents { get; set; }
        public string Discriminator { get; set; }

        public Registry() {
            this.DateAdded = DateTime.Now;
            this.Date = DateTime.Today;
            this.Documents = new List<Document>();
        }
    }
}

File: ClientApp/src/app/shared/models/registry.ts
import { Document } from './document';

export class Registry {
  registryID: number;
  indicatorID: number;
  name: string;
  dateAdded: Date;
  date: Date;
  documents: Document [] = new Array<Document>();
  discriminator: string;
  quantity: number;
  percent: number;

  constructor(dateAdded?: Date, date?: Date, name?: string,
     quantity?: number, percent?: number ) {
    this.date = date;
    this.dateAdded = dateAdded;
    this.name = name;
    this.quantity = 0;
    this.percent = 0;
  }
}
